### PR TITLE
Added Gloo as a backend option

### PR DESCRIPTION
**Pull request by sayantan1410**
Fix #201 
### What is the purpose of this pull request?
- [X] New feature

Let me know if I something needs to be changed.

### CODE REVIEW

**Comment by sayantan1410** [comment]
@vfdev-5 I ran this `bash scripts/run_code_style.sh lint` but it did not make any changes, while the ci/lint test is failing, can you tell me why ?

**Comment by vfdev-5** [comment]
@sayantan1410 thanks for the PR, almost good. Two things to fix:
- run prettier to fix ci/lint (https://github.com/pytorch-ignite/code-generator/runs/4934201901?check_suite_focus=true#step:9:9)
  - I think our contributing guide lacks this info, cc @jdcjeff
- replace all occurencies of nccl by tags+backend in all readmes
  - for example, https://github.com/pytorch-ignite/code-generator/pull/203/files#diff-ae96dfae20580bd8b1957960a4b26c1eced4361fd511af8a0730cd3ce3fc92f2L47

File: src/templates/template-vision-segmentation/README.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ python main.py  \
   --node_rank 0 \
   --master_addr #:::= it.master_addr :::# \
   --master_port #:::= it.master_port :::# \
-  --backend nccl
+  --backend #:::= backend :::#
```
Comment (sayantan1410) [inline src/templates/template-vision-segmentation/README.md:80]: In line 58 whenever, I am trying to replace the nccl string with `#:::= backend :::#`, the application is failing and I couldn't figure out why.

File: src/templates/template-text-classification/README.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ python -m torch.distributed.launch \
   --master_addr #:::= it.master_addr :::# \
   --master_port #:::= it.master_port :::# \
   --use_env main.py \
-  --backend nccl
+  --backend #:::= backend :::#
```
Comment (vfdev-5) [inline src/templates/template-text-classification/README.md:34]: Can you try `#:::= it.backend :::#`, I think there should be a typo on `#:::= nproc_per_node :::#` which should be `#:::= it.nproc_per_node :::#`

@ydcjeff why we have `#:::= nproc_per_node :::#` and just above `#::: if (it.nproc_per_node) { :::#` ?

**Comment by ydcjeff** [comment]
@sayantan1410 if you have installed npm packages, you can run `pnpm run fmt` to auto format.

File: src/templates/template-text-classification/README.md (hunk 2 above)
Comment (ydcjeff) [reply src/templates/template-text-classification/README.md:34]: Yea, i think it should include `it.`.

**Review by vfdev-5 — APPROVED** [review]
Thanks a lot for the PR @sayantan1410 !
LGTM

**Comment by sayantan1410** [comment]
@ydcjeff thanks that worked !!
@vfdev-5 thanks for merging !!

**Pull request merged**